### ISSUE TITLE
Python 3.6 is EOL

### ISSUE THREAD

**Issue by systemcatch** [arrow-py/arrow#1083]
Python 3.6 reached EOL in December, here is the current split by version for arrow usage.

| category | percent | downloads |
|:---------|--------:|----------:|
| 3.7      |  38.26% | 1,910,775 |
| 3.8      |  32.76% | 1,635,938 |
| 3.6      |  12.85% |   641,712 |
| 3.9      |   6.48% |   323,713 |
| null     |   3.70% |   185,035 |
| 2.7      |   3.07% |   153,354 |
| 3.10     |   2.26% |   112,810 |
| 3.5      |   0.51% |    25,236 |
| 3.4      |   0.11% |     5,477 |
| 3.11     |   0.01% |       404 |
| Total    |         | 4,994,454 |

Date range: 2021-12-01 - 2021-12-31

**Comment by krisfremen** [comment]
Thanks for the version numbers breakdown @systemcatch

**Comment by talhajunaidd** [comment]
Is there any technical reason to remove the support for python 3.6?

**Comment by jadchaar** [comment]
Nope not as of now. Most new language features are backported back to 3.6 so I doubt we will drop it anytime soon. I think we should wait at least a year before dropping support to give teams time to migrate to newer versions of Python.

**Comment by systemcatch** [comment]
@jadchaar do you think we should do a deprecation warning for 3.6 on install like we did for 2.7?

**Comment by jadchaar** [comment]
Eventually we should, but I think until we have a concrete path to deprecation and an exact timeline I don’t know if it’s worth throwing a deprecation warning which annoyed some users during 2.7 deprecation.

I think we should give users some time to migrate away from 3.6 since it just went EOL. And down the line we can issue a warning once the numbers trend downward and supporting 3.6 becomes an annoyance. Until we encounter issues with 3.6 and other major packages drop support I don’t think we should make a move on this.

**Comment by systemcatch** [comment]
Agreed I think that is the best course of action for now, I will keep posting the usage stats every month so we stay up to date.

**Comment by krisfremen** [comment]
@systemcatch hey, would you be able update the stats to see if we should consider making a decision on this?

**Comment by systemcatch** [comment]
@krisfremen no problem, here are the stats for April.

| category | percent | downloads |
|:---------|--------:|----------:|
| 3.8      |  33.20% | 2,175,467 |
| 3.7      |  32.77% | 2,147,326 |
| 3.9      |  11.80% |   773,300 |
| 3.6      |   9.75% |   639,097 |
| 3.10     |   5.29% |   346,789 |
| null     |   4.51% |   295,713 |
| 2.7      |   2.36% |   154,787 |
| 3.5      |   0.22% |    14,484 |
| 3.4      |   0.06% |     4,254 |
| 3.11     |   0.02% |     1,067 |
| 3.3      |   0.00% |         5 |
| Total    |         | 6,552,289 |

Date range: 2022-04-01 - 2022-04-30

3.6 usage has dropped from 12.85% to 9.75% since December.
